scaffolder-templates/advanced-workflow/template.yaml

```diff
@@ -162,7 +162,74 @@ spec:
                 - persistencePSQLServiceNamespace
                 - persistencePSQLServicePort
                 - persistencePSQLDatabaseName
+    - title: Create Quay Repository
+      required:
+        - quayCreateNewRepo
+        - CI
+      properties:
+        quayCreateNewRepo:
+          title: Create a new Quay repository with the provided name? (You must select a CI/CD Method in the Build Environment)
+          type: string
+          default: "Yes"
+          enum:
+            - "Yes"
+            - "No"
+      dependencies:
+        quayCreateNewRepo:
+          oneOf:
+            - properties:
+                quayCreateNewRepo:
+                  const: "No"
+            - properties:
+                quayCreateNewRepo:
+                  const: "Yes"
+                token:
+                  title: Quay AuthToken
+                  type: string
+                  description: Bearer token used for authorization
+                  ui:widget: password
+                visibility:
+                  title: New quay Repository Visiblity
+                  type: string
+                  description: Visibility setting for the created repository, either public or private
+                  ui:widget: select
+                  enum: ["public", "private"]
+                repoKind:
+                  title: New Quay Repository Kind
+                  type: string
+                  description: The created repository kind either image, application or none
+                  ui:widget: select
+                  enum: ["image", "application"]
+                newRepoDescription:
+                  title: New Quay Repository Description
+                  type: string
+                  description: The repository desription
+                baseUrl:
+                  title: Quay Base URL
+                  type: string
+                  description: URL of your quay instance, set to "https://quay.io" by default
```

Comment:
Do we even need that? Is it possible to have quay repository that are not on quay.io? 
When custom registry will be implemented, maybe it will be possible to have a registry of type `quay`

Reply:
I think it's for quay instances that aren't quay.io, not for external reg

Reply:
do you have an example? I never encountered such thing but I am keen to acknowledge their existence

Reply:
>I think it's for quay instances that aren't quay.io, not for external reg
The quay API docs use the term "https://quay-server.example.com/" to refer to the base URL

Ok, I am not familiar with quay, but from what I understand, anyone could deploy a quay registry (hence the `kind` of the custom registry), so let's keep this field

In the PR to use custom registry, we will need to take care of the wording; the custom registry will refer to another kind of registry than `quay`

Reply:
I never saw it in person, but [here ](https://www.redhat.com/en/technologies/cloud-computing/quay#:~:text=Ways%20to%20run%20Red%20Hat%20Quay)it looks like it exists